utils.js

```diff
@@ -322,9 +322,16 @@ export class InjectionsHandler extends BasicHandler {
         if (!(original instanceof Function))
             throw new Error(`Virtual function ${name}() is not available for ${object}`);
 
-        object[name] = function (...args) {
-            return injectedFunction.call(this, original, ...args);
-        };
+        if (original.constructor.name === 'AsyncFunction') {
+            // eslint-disable-next-line require-await
+            object[name] = async function (...args) {
+                return injectedFunction.call(this, original, ...args);
+            };
```

Comment:
So, if you're declaring this `async` and `injectedFunction` that would be meant to await, so I think you can just drop the async and return the promise as it is no?

Reply:
I expect #2110 will be solved either by #2166 or by your own commit you said you had, or both. So we don't need to continue here.